### PR TITLE
zh-CN: New translation for 'abi-stability.md'

### PR DESCRIPTION
**Pull request by ghost**
Ref: https://github.com/nodejs/nodejs.org/pull/1828

@nodejs/nodejs-cn

### CODE REVIEW

File: locale/zh-cn/docs/guides/abi-stability.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+# ABI 的稳固性
+
+## 简介
+应用程序二进制接口（ABI）对于程序而言是调用函数，从其它已编译的程序中使用数据结构的一种方式。它是应用编程接口（API）的已编译版本。换句话说：描述类、函数、数据结构、枚举和常量的头文件，它们使应用程序能够执行所需的任务方法是编译到一组地址和预期的参数值和内存。编译 ABI 提供程序的结构大小和布局。
+
+编译使用 ABI 的应用程序必须使得可用地址、预期参数值和内存结构大小和布局与 ABI 提供程序编译的那些协议一致。这通常是通过对 ABI 提供程序提供的标头进行编译来实现的。
+
+由于 ABI 的提供者和 ABI 的用户可以在不同的时间使用不同的编译器进行编译，因此确保 ABI 兼容性的部分责任在于编译器。编译器的不同版本（可能由不同的供应商提供）都必须从具有特定内容的头文件中生成相同的 ABI，并且必须使用 ABI 访问给定标头中描述的 API 来生成应用程序的代码。这些代码根据程序头部的描述转换得来。现代编译器具有相当好的跟踪记录，不破坏它们编译的应用程序的 ABI 兼容性。
+
+保持并确保 ABI 兼容性的责任在于维护标头文件的团队，后者提供 API；在编译后使得在 ABI 中保持稳定。对头文件进行更改是可以的，但必须密切跟踪更改的性质，以确保在编译时 ABI 不会更改，从而使 ABI 的现有用户与新版本不兼容。
```
Comment (terry-fei) [inline locale/zh-cn/docs/guides/abi-stability.md]: ； 应该是，
感觉应该是与新版本兼容

File: locale/zh-cn/docs/guides/abi-stability.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+* JavaScript 语言自其早期就一直与自身兼容，而执行 JavaScript 代码的引擎 ABI 则随 Node.js 的每个主要版本而变化。这意味着在 JavaScript 中完全写入 Node.js 包中所包含的应用程序不需要重新编译和重新安装，或者作为新的主要版本的 Node.js 被丢弃到运行此类应用程序的生产环境中。相比之下，如果应用程序依赖于包含本机插件的包，仅将 Node.js 的新主版本引入到生产环境中就必须重新编译、重安装和部署应用程序。这种差距在包含本机插件的 Node.js 包和完全写入 JavaScript 的 Nodejs 之间，增加了依赖于本机插件的生产系统的维护负担。
+
+* 其他项目已经开始产出 JavaScript 接口，这基本上是 node.js 的替代实现。因为这些项目通常是建立在不同 JavaScript 引擎而不是 V8；他们的本机插件必须采取不同的结构和使用不同的 API。不过，在 Node.js 的 JavaScript API 的不同实现中，使用本机插件的单个 API 将允许这些项目利用 Node.js 包中累积的 JavaScript 软件包的生态系统。
+
+* Node.js 可能在将来包含不同的 JavaScript 引擎。这意味着, 对外所有 Node.js 接口将保持不变，但 V8 头文件将不存在。如果某个 API 没有首先被 Node.js 引擎提供，并且没有被本地插件采用。那么这一步将导致 Node.js 生态系统的中断——特别是本机插件的破坏。
+
+对于这些收束，Node.js 已在版本 8.6.0 中引入了 N API 并将其标记为项目的稳定组件——如 Node.js 8.12.0。这些 API 在头文件 [`node_api.h`][] 和 [`node_api_types.h`][]中定义，并提供跨越 Node.js 主要版本边界的前向兼容性保证。保证可以说明如下:
+
+
+**n API 的给定版本 *n* 将在其发布的 Node.js 的主要版本和所有后续版本的 Node.js 中提供，包括后续的主要版本。**
```
Comment (willin) [inline locale/zh-cn/docs/guides/abi-stability.md]: **N-API 的给定版本 *N* 将在其发布的 Node.js 的主要版本和所有后续版本的 Node.js 中提供，包括后续的主要版本。**

```md
**N-API 的给定版本 *N* 将在其发布的 Node.js 的主要版本和所有后续版本的 Node.js 中提供，包括后续的主要版本。**
```

**Review by willin — APPROVED** [review]
LGTM

File: locale/zh-cn/docs/guides/abi-stability.md (hunk 1 above)
Comment (stevemao) [reply locale/zh-cn/docs/guides/abi-stability.md]: Should be:
`从而不会使 ABI 的现有用户与新版本不兼容`

File: locale/zh-cn/docs/guides/abi-stability.md (hunk 2 above)
Comment (ghost) [reply locale/zh-cn/docs/guides/abi-stability.md]: 你的意思是大小写问题？

Comment (willin) [reply locale/zh-cn/docs/guides/abi-stability.md]: 对

Comment (stevemao) [reply locale/zh-cn/docs/guides/abi-stability.md]: @Maledong there is also a dash in between: `N-API`

**Review by haoxins — APPROVED** [review]

File: locale/zh-cn/docs/guides/abi-stability.md (hunk 1 above)
Comment (ghost) [reply locale/zh-cn/docs/guides/abi-stability.md]: @stevemao：Fixed

**Comment by Maledong** [comment]
@coderafei , @stevemao , @willin：Please have a review :)

File: locale/zh-cn/docs/guides/abi-stability.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
+---
+title: ABI Stability
+layout: docs.hbs
+---
+
+# ABI 的稳固性
```
Comment (marswong) [inline locale/zh-cn/docs/guides/abi-stability.md]:
```suggestion
# ABI 的稳定性
```

File: locale/zh-cn/docs/guides/index.md [hunk 4]
```diff
@@ -29,3 +29,4 @@ layout: docs.hbs
 - [流的背压](backpressuring-in-streams/)
 - [Postmortem 中的主要模块](domain-postmortem/)
 - [如何发布 N-API 包](publishing-napi-modules/)
+- [ABI 的稳固性](abi-stability/)
```
Comment (marswong) [inline locale/zh-cn/docs/guides/index.md]:
```suggestion
- [ABI 的稳定性](abi-stability/)
```

File: locale/zh-cn/docs/guides/abi-stability.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+## 简介
+应用程序二进制接口（ABI）对于程序而言是调用函数，从其它已编译的程序中使用数据结构的一种方式。它是应用编程接口（API）的已编译版本。换句话说：描述类、函数、数据结构、枚举和常量的头文件，它们使应用程序能够执行所需的任务方法是编译到一组地址和预期的参数值和内存。编译 ABI 提供程序的结构大小和布局。
```
Comment (marswong) [inline locale/zh-cn/docs/guides/abi-stability.md]:
```suggestion
应用程序二进制接口（ABI）是程序通过调用函数使用其它编译程序中的数据结构的一种方式。它是应用程序接口（API）的编译版本。换句话说：描述类、函数、数据结构、枚举和常量的头文件，它们使应用程序能够执行所需的任务方法是编译到一组地址和预期的参数值和内存。编译 ABI 提供程序的结构大小和布局。
```

File: locale/zh-cn/docs/guides/abi-stability.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+编译使用 ABI 的应用程序必须使得可用地址、预期参数值和内存结构大小和布局与 ABI 提供程序编译的那些协议一致。这通常是通过对 ABI 提供程序提供的标头进行编译来实现的。
```
Comment (marswong) [inline locale/zh-cn/docs/guides/abi-stability.md:11]:
```suggestion
使用 ABI 的应用程序必须编译，使得可用地址、预期参数值和内存结构大小和布局与 ABI 提供方提供的编译程序一致。这通常通过对 ABI 提供方提供的头文件进行二次编译来实现。
```

**Review by terry-fei — APPROVED** [review]

File: locale/zh-cn/docs/guides/abi-stability.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+由于 ABI 的提供者和 ABI 的用户可以在不同的时间使用不同的编译器进行编译，因此确保 ABI 兼容性的部分责任在于编译器。编译器的不同版本（可能由不同的供应商提供）都必须从具有特定内容的头文件中生成相同的 ABI，并且必须使用 ABI 访问给定标头中描述的 API 来生成应用程序的代码。这些代码根据程序头部的描述转换得来。现代编译器具有相当好的跟踪记录，不破坏它们编译的应用程序的 ABI 兼容性。
```
Comment (marswong) [inline locale/zh-cn/docs/guides/abi-stability.md]:
```suggestion
由于 ABI 提供方和 ABI 用户可能会在不同时刻使用不同版本的编译器进行编译，因此确保 ABI 兼容性的部分责任在于编译器。不同版本的编译器（可能由不同的供应商提供）都必须从具有特定内容的头文件生成相同的 ABI，并且必须使用 ABI（根据约定使用给定头文件描述的 API）生成应用程序代码。现代编译器具有相当好的跟踪记录，不破坏它们编译的应用程序的 ABI 兼容性。
```

File: locale/zh-cn/docs/guides/abi-stability.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+保持并确保 ABI 兼容性的责任在于维护标头文件的团队，后者提供 API；在编译后使得在 ABI 中保持稳定。对头文件进行更改是可以的，但必须密切跟踪更改的性质，以确保在编译时 ABI 不会更改，从而不会导致 ABI 的现有用户与新版本不兼容。
```
Comment (marswong) [inline locale/zh-cn/docs/guides/abi-stability.md:15]: 标头（文件）建议统一用头文件

File: locale/zh-cn/docs/guides/abi-stability.md [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+## ABI 在 Node.js 中的稳固性
```
Comment (marswong) [inline locale/zh-cn/docs/guides/abi-stability.md]: 稳固性建议统一用稳定性

**Review by marswong — CHANGES_REQUESTED** [review]
建议边机翻边检查语法问题，不然校对成本太高🙈

File: locale/zh-cn/docs/guides/abi-stability.md (hunk 9 above)
Comment (stevemao) [reply locale/zh-cn/docs/guides/abi-stability.md]: I agree. Never heard of `稳固性` before

**Comment by Maledong** [comment]
@marswong：I'm manually translating instead of using auto-translation.

Of course, I'll fix what you all suggest.

**Comment by Maledong** [comment]
@marswong: Fixed all, many thanks!

**Review by stevemao — APPROVED** [review]
Let's ship it!
Any other improvements just make another PR :)

**Comment by Maledong** [comment]
@stevemao：No shit any more :)

I'll leave this for a while and try my best to adpot your suggestions into the change, if forwarding to the same issue or related topic.

**Pull request merged**